python/BioSimSpace/Sandpit/Exscientia/FreeEnergy/_restraint_search.py

```diff
@@ -0,0 +1,1113 @@
+            raise ValueError("'ignore_warnings' must be of type 'bool'.")
+        self._ignore_warnings = ignore_warnings
+
+        if not isinstance(show_errors, bool):
+            raise ValueError("'show_errors' must be of type 'bool'.")
+        self._show_errors = show_errors
+
+        # Check that the map is valid.
+        if not isinstance(property_map, dict):
+            raise TypeError("'property_map' must be of type 'dict'.")
+        self._property_map = property_map
+
+        # Create fake instance methods for 'analyse'. These
+        # pass instance data through to the staticmethod versions.
+        self.analyse = self._analyse
+
+        # Initialise the process.
+        self._initialise_process(self._system, gpu_support, **kwargs)
+
+    def start(self):
+        """Start the simulation."""
+        self._process.start()
+
+    def wait(self):
+        """Wait for the simulation to finish."""
+        self._process.wait()
+
+    def kill(self):
+        """Kill the process."""
+        self._process.kill()
+
+    def isRunning(self):
+        """Check if the process is running."""
+        self._process.isRunning()
+
+    def workDir(self):
+        """Return the working directory.
+
+           Returns
+           -------
+
+           work_dir : str
+               The path of the working directory.
+        """
+        return self._work_dir
+
+    def _analyse(self, rest_type='Boresch',
+                method='MDRestraintsGenerator',
+                append_to_lig_selection="",
+                recept_selection_str='protein and name CA C N',
+                cutoff=10, # In Angstrom
+                block='AUTO'):
+        """Analyse trajectory and select restraints which best mimic strongest
+        receptor-ligand interactions.
+
+           Parameters
+           ----------
+
+           rest_type: str
+               The type of restraints to select (currently only Boresch is available).
+               Default is 'Boresch'.
+
+           method: str
+               The method to use to derive the restraints. Currently only 'MDRestraintsGenerator'
+               is supported.
+
+           append_to_lig_selection: str
+               Appends the supplied string to the default atom selection which chooses
+               the atoms in the ligand to consider as potential anchor points. The default
+               atom selection is f'resname {ligand_resname} and not name H*'. Uses the
+               mdanalysis atom selection language. For example, 'not name O*' will result
+               in an atom selection of f'resname {ligand_resname} and not name H* and not
+               name O*'.
+
+           recept_selection_str: str
+               The selection string for the atoms in the receptor to consider
+               as potential anchor points. The default atom selection is
+               'protein and name CA C N'. Uses the mdanalysis atom selection
+               language.
```

Comment:
Just to note that Sire 2023 will also have trajectory handling (not sure what formats, yet) and will be able to search/return properties from different frames. This might make it easier to provide this functionality natively. (Definitely out of scope for now.)